lightwood/encoders/time_series/helpers/rnn_helpers.py

```diff
@@ -2,58 +2,84 @@
 
 import torch
 import torch.nn as nn
-import torch.nn.functional as F
 
-import numpy as np
 
-
-def tensor_from_series(series, device):
-    if type(series) != type([]):
-        series = str(series).split(' ')
+def tensor_from_series(series, device, n_dims=None, pad_value=0.0, max_len=64):
+    """
+    :param series: list of lists, corresponds to time series: [[x1_1, ..., x1_n], [x2_1, ..., x2_n], ...]
+                   the series is zero-padded on each axis so that all dimensions have equal length
+    :param device: computing device that PyTorch backend uses
+    :param n_dims: if not None, will zero-pad dimensions until series_dimensions = n_dims
+    :param pad_value: value to pad each dimension in the time series, if needed
+    :param max_len: length to pad or truncate each time_series
+    :return: series as a tensor ready for model consumption, shape (1, ts_length, n_dims)
+    """
+    if isinstance(series[0], torch.Tensor):
+        return series[0]
 
     float_series = []
-    for ele in series:
-        try:
-            float_series.append(float(ele))
-        except:
-            logging.warning(f'Weird element encountered in timeseries: {ele} !')
-            float_series.append(0)
-
-    return torch.tensor(float_series, dtype=torch.float, device=device).view(-1, 1, 1, 1).float()
+    for dimn in series:
+        dimn_series = []
+        if isinstance(dimn[0], str):
+            dimn = [i for i in dimn[0].split(" ")]
+        for ele in dimn:
+            try:
+                dimn_series.append(float(ele))
+            except Exception as _:
+                logging.warning(f'Weird element encountered in timeseries: {ele} !')
+                dimn_series.append(0)
+        float_series.append(dimn_series)
+
+    # timestep padding and truncating
+    timesteps = max([len(dimn) for dimn in float_series])
+    for i in range(len(float_series)):
+        # for _ in range(max(0, timesteps - len(float_series[i]))):
+        for _ in range(max(0, max_len - timesteps)):
+            float_series[i].append(pad_value)
```

Comment:
Those this padding algorithm work ?

It seem incorrect to me, since `max_len` and `timesteps` will always be the same, but float_series[0] and float_series[1] could be of different lengths.

Also, in the previous itteration of `for _ in range(max(0, timesteps - len(float_series[i])))` the `max` is still incorrect, since when the subtraction is < 0 we should be removing extra elements instead of padding.

Reply:
Fixed.